src/core/control/Control.cpp

```diff
@@ -1779,7 +1779,28 @@ void Control::undoRedoPageChanged(PageRef page) {
     }
 }
 
-void Control::selectTool(ToolType type) {
+void Control::handleSelectToolAction(ToolType type) {
+    // keep text-selection when switching from text to seletion tool
+    auto oldTool = getToolHandler()->getActiveTool();
+    if (oldTool && win
+                && (type == TOOL_SELECT_RECT || type == TOOL_SELECT_REGION || type == TOOL_SELECT_OBJECT)
+                && oldTool->getToolType() == ToolType::TOOL_TEXT
+                && !(this->win->getXournal()->getTextEditor()->getText()->getText().empty())) {
+        auto xournal = this->win->getXournal();
+        Text* textobj = std::move(xournal->getTextEditor()->getText());
```

Comment:
nit: `xournal->getTextEditor()->getText()` returns a pointer, so a move is unnecessary.
```suggestion
        Text* textobj = xournal->getTextEditor()->getText();
```